### PR TITLE
JFG-1002 & 1004: make QueryProvider optional for Jagger tests; add property chassis.master.using.builders=false to test configuration

### PR DESCRIPTION
**Pull request by anton-antonenko**
JFG-1002: make QueryProvider optional for Jagger tests;
JFG-1004: add property chassis.master.using.builders=false to test configuration

### CODE REVIEW

File: chassis/core/src/main/java/com/griddynamics/jagger/invoker/PairSupplierFactoryLoadBalancer.java [hunk 1]
```diff
@@ -30,8 +30,8 @@ public void setPairSupplierFactory(PairSupplierFactory<Q, E> pairSupplierFactory
     }
 
     protected PairSupplier<Q, E> getPairSupplier(){
-        if (endpointProvider == null || queryProvider == null){
-            throw new NullPointerException("Init query and endpoint providers!");
+        if (endpointProvider == null){
```
Comment (SokolAndrey) [inline chassis/core/src/main/java/com/griddynamics/jagger/invoker/PairSupplierFactoryLoadBalancer.java]: Maybe here should the same check for queryProvider?

File: chassis/core/src/main/java/com/griddynamics/jagger/invoker/v2/SpringBasedHttpClient.java [hunk 2]
```diff
@@ -103,6 +107,13 @@ public JHttpResponse execute(JHttpEndpoint endpoint, JHttpQuery query) {
         return mapToJHttpResponse(responseEntity);
     }
 
+    public JHttpResponse execute(JHttpEndpoint endpoint) {
+        URI endpointURI = endpoint.getURI();
+        RequestEntity requestEntity = mapToRequestEntity(endpointURI);
+        ResponseEntity responseEntity = restTemplate.exchange(endpointURI, HttpMethod.GET, requestEntity, byte[].class);
+        return mapToJHttpResponse(responseEntity);
```
Comment (dlatnikov) [inline chassis/core/src/main/java/com/griddynamics/jagger/invoker/v2/SpringBasedHttpClient.java]: don't like code duplicated in he two different methods

File: test/jagger-test/src/main/resources/profiles/ci-distributed/environment-master.properties [hunk 3]
```diff
@@ -39,3 +39,5 @@ test.service.endpoints=localhost
 test.configuration.tickinterval=500
 test.configuration.tps=500
 test.configuration.duration=5m
+
+chassis.master.using.builders=false
```
Comment (dlatnikov) [inline test/jagger-test/src/main/resources/profiles/ci-distributed/environment-master.properties]: are you sure it is the only place. there are two xml configured tests

File: chassis/core/src/main/java/com/griddynamics/jagger/invoker/v2/SpringBasedHttpClient.java (hunk 2 above)
Comment (anton-antonenko) [inline chassis/core/src/main/java/com/griddynamics/jagger/invoker/v2/SpringBasedHttpClient.java]: It's not duplicated. 
It's better to handle cases with query and without it in different methods.

File: chassis/core/src/main/java/com/griddynamics/jagger/invoker/RoundRobinPairSupplierFactory.java [hunk 4]
```diff
@@ -31,29 +31,32 @@
 public class RoundRobinPairSupplierFactory<Q, E> implements PairSupplierFactory<Q, E> {
     @Override
     public PairSupplier<Q, E> create(Iterable<Q> queries, Iterable<E> endpoints) {
-
-        LinkedList<Pair<Q, E>> tempList = new LinkedList<Pair<Q, E>>();
+        LinkedList<Pair<Q, E>> tempList = new LinkedList<>();
```
Comment (a-badaev) [inline chassis/core/src/main/java/com/griddynamics/jagger/invoker/RoundRobinPairSupplierFactory.java]: can be replaced by ArrayList

**Pull request merged**